lookout/core/manager.py

```diff
@@ -93,16 +90,23 @@ def process_push_event(self, request: PushEvent) -> EventResponse:  # noqa: D401
         Callback for push events invoked by EventListener.
         """
         ptr = ReferencePointer.from_pb(request.commit_revision.head)
+        data_service = self._data_service
         for analyzer in self._analyzers:
             if analyzer.model_type == DummyAnalyzerModel:
                 continue
-            self._log.debug("training %s", analyzer.name)
             try:
                 mycfg = self._protobuf_struct_to_dict(request.configuration[analyzer.name])
             except (KeyError, ValueError):
                 mycfg = {}
+            model = self._get_model(analyzer, ptr.url)
+            if model is not None:
+                must_train = analyzer.check_training_required(model, ptr, mycfg, data_service)
```

Comment:
just want to remind you about this issue: https://github.com/src-d/lookout-sdk-ml/issues/17
You expect train config in `check_training_required` but passes analyze one.

Reply:
Yes. There is not much to be done here for now because Lookout does not know anything about stateful analyzers. So this should be handled separately by each analyzer.